web/src/beta/features/PluginPlayground/Code/utils.ts

```diff
@@ -0,0 +1,30 @@
+const REEARTH_HTML_INJECTION_PATTERN =
+  /reearth\.(ui|modal|popup)\.show\(\s*`([^]*?)`\s*\);/g;
+
+export const extractHtmlFromSourceCode = (sourceCode: string) => {
+  const match = REEARTH_HTML_INJECTION_PATTERN.exec(sourceCode);
+  return match?.[2];
+};
+
+export const injectHtmlIntoSourceCode = (
+  htmlSourceCode: string,
+  sourceCode: string
+) => {
+  return sourceCode.replace(
+    REEARTH_HTML_INJECTION_PATTERN,
+    (_, type) => `reearth.${type}.show(\`${htmlSourceCode}\`);`
+  );
```

Comment:
Same as above, there could be mulitple modal.show popup.show ui.show exist within one source code.